src/Maui/Pages/Combat.razor

```diff
@@ -1,83 +1,73 @@
-﻿<MudGrid>
-        <MudItem xs="12">
+﻿
+<MudGrid>
+    <MudItem xs="12">
         <MudAutocomplete T="string" Label="Creature" @bind-Value="@returnCreature" SearchFunc="@Filter" Variant="Variant.Outlined" Immediate="true" ResetValueOnEmptyText="true" />
-    <MudButton Variant="Variant.Filled" Color="Color.Primary" OnClick="InitiateCreature">Add To Combat</MudButton>
+        <MudButton Variant="Variant.Filled" Color="Color.Primary" OnClick="InitiateCreature">Add To Combat</MudButton>
         <MudButton Variant="Variant.Outlined" Color="Color.Secondary" OnClick="InitiativeRoll">Initiative Roll</MudButton>
     </MudItem>
-     <MudItem xs="12">
-        @foreach (InitiatedCreature activeCreature in InitiatedCreatures)
+
+    @foreach (InitiatedCreature activeCreature in InitiatedCreatures)
+    {
+        <MudItem lg="3" xs="12">
+            <MudCard>
+                <MudCardHeader>
+                    <CardHeaderAvatar>
+                        <MudAvatar Image="" Size="Size.Large" />
+                    </CardHeaderAvatar>
+                    <CardHeaderContent>
+                        <MudText Typo="Typo.h5">@activeCreature.Creature.Name</MudText>
+                        <MudText Typo="Typo.caption"><b>CR:</b>@activeCreature.Creature.cr  <b>HP:</b>@activeCreature.HitPoints</MudText>
+
+                    </CardHeaderContent>
+                </MudCardHeader>
+                <MudCardContent>
+                    <MudText Typo="Typo.caption"><b>Initiative Roll:</b>:@activeCreature.InititiveRoll</MudText>
+                    <MudButton Text="Details" Color="Color.Secondary" OnClick="@(() => OpenDrawer(activeCreature))">Details</MudButton>
+                </MudCardContent>
+            </MudCard>
+        </MudItem>
+    }
+    <MudDrawer Width="100%" Height="80%" @bind-Open="@open" Anchor="Anchor.Bottom" Elevation="1" Variant="@DrawerVariant.Temporary">
+        <MudDrawerHeader>
+            <MudText Typo="Typo.h6">@clickedCreature.Creature.Name</MudText>
```

Comment:
Did you want to use the CreatureDetail component? That way we don’t have to recreate the wheel every time.

The CreatureDetail component currently doesn’t have abilities and actions, tho.

Reply:
Not really, I think each selected creature needs to be different